### ISSUE TITLE
Issue when saving

### ISSUE THREAD

**Issue by yulianglukhenko** [infinitered/thesis-phoenix#21]
Env: dev
Version: 0.0.15

This is the ThesisAuth module:

```
defmodule IcityMortgage.ThesisAuth do
  @moduledoc """
  Contains functions for handling Thesis authorization.
  """

  def page_is_editable?(conn) do
    user = Guardian.Plug.current_resource(conn)
    IO.inspect user
    IO.inspect user.role
    IO.inspect user && user.role == :admin
    user && user.role == :admin
  end
end
```

Thesis editor is visible after logging in = great: 

![](http://d.pr/i/1f5YK+)

After editing a content area and saving, I get an error:

![](http://d.pr/i/14TGe+)

As you can see, inspect for user returns `nil`. Does the session not get passed into the Thesis scope? 

@ryanlntn @jamonholmgren

**Comment by yulianglukhenko** [comment]
More info: heres what the application router pipelines look like: ![](http://d.pr/i/vOFF+)

**Comment by yulianglukhenko** [comment]
More info:

A regular site request (eg /) contains a few Guardian references in the conn: 
![](http://d.pr/i/1fYyx+)

However, the `/thesis/update` request contains only one Guardian reference:
![](http://d.pr/i/1hHY4+)

**Comment by yulianglukhenko** [comment]
My guess is Thesis doesn't see the session because we aren't passing through the browser_session pipeline to the "/thesis" scope ??? 

@jamonholmgren

**Comment by yulianglukhenko** [comment]
Adding browser_session to the pipe_through in Thesis fixes the issue: 
![](http://d.pr/i/1bEiV+)

I'm not sure if this is something that needs to be added to Thesis or changed on my app.

Please advise.

@jamonholmgren @ryanlntn

**Comment by jamonholmgren** [comment]
Working on this. I think we'll need them to add the pipeline manipulations in the `can_edit_page?/1` function.

**Comment by jamonholmgren** [comment]
Addressed in #26, relevant README section: https://github.com/infinitered/thesis-phoenix/blob/master/README.md#authorization
